### PR TITLE
moved enum_net.hpp, peer_class_set.hpp and proxy_base.hpp to aux

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

**Comment by arvidn** [comment]
https://cirrus-ci.com/task/4664715794186240?command=build_cmake#L136

**Review by arvidn — APPROVED** [review]

**Comment by arvidn** [comment]
https://cirrus-ci.com/task/5681668419747840?command=enum_if#L12

**Comment by arvidn** [comment]
not sure what this is. it doesn't look related to your change: https://github.com/arvidn/libtorrent/pull/5864/checks?check_run_id=1705041154#step:4:159

**Comment by aldenml** [comment]
it's not related to this change. It seems that the fix to compile with boost-1.75 is not that robust

File: test/enum_if.cpp [hunk 1]
```diff
@@ -24,34 +24,34 @@ std::string operator "" _s(char const* str, size_t len) { return std::string(str
...
+		+ ((f & aux::if_flags::lower_up) ? "LWR_UP "_s : ""_s)
+		+ ((f & aux::if_flags::dormant) ? "DORMANT "_s : ""_s)
 		;
 }
 
 char const* print_state(aux::if_state const s)
 {
 	switch (s)
 	{
-		case if_state::up: return "up";
```
Comment (arvidn) [inline test/enum_if.cpp:48]: would you mind sticking this at the top of the function instead?

```
using if_state = aux::if_state;
```

Comment (arvidn) [reply test/enum_if.cpp:48]: (and the function above too)

Comment (aldenml) [reply test/enum_if.cpp:48]: sure, looks better

**Comment by arvidn** [comment]
https://cirrus-ci.com/task/5717914252738560?command=enum_if#L10

It seems this is a namespace, so you could pull it in by:
```
namespace if_flags = aux::if_flags;
```

**Comment by aldenml** [comment]
cirrus finally is green! the issue with boost-1.75 and macOS persists :(

**Pull request merged**